Sources/TUSKit/TUSClient.swift

```diff
@@ -103,7 +103,7 @@ public final class TUSClient {
     /// - Important: The client assumes by default that your server implements the Creation TUS protocol extension. If your server does not support that,
     ///   make sure that you provide an empty array in the `supportExtensions` parameter.
     /// - Throws: File related errors when it can't make a directory at the designated path.
-    public init(server: URL, sessionIdentifier: String, storageDirectory: URL? = nil, session: URLSession = URLSession.shared, chunkSize: Int = 500 * 1024, supportedExtensions: [TUSProtocolExtension] = [.creation]) throws {
+    public init(server: URL, sessionIdentifier: String, storageDirectory: URL? = nil, session: URLSession = URLSession.shared, chunkSize: Int = 500 * 1024, supportedExtensions: [TUSProtocolExtension] = []) throws {
```

Comment:
Apologies for taking a while to pick this review back up. But I think we should not change this default argument. Users of TUSKit will be reliant on public method behavior not changing so this would introduce a breaking change.

Was there a particular reason why this should be an empty array in your implementation?

Reply:
This is because the current TUSKit API design does not guarantee that the initial data will be correctly validated on subsequent requests

Reply:
I'm not entirely sure what you mean by that. Is the idea that instead of having this default argument where we assume the server supports creation you'd rely on asking the server for its extensions and obtain the default value that way instead? That would essentially make the entire argument obsolete then right?

My main issue here is that this is a break of public API which I'd like to avoid.